test/regfox/regfox_api.test.js

```diff
@@ -136,4 +154,36 @@ describe('regfox_api', () => {
       return login('anything', 'none').should.eventually.be.rejected;
     });
   });
+
+  describe('getRegistrationInfo', () => {
+    const getResponse = stub();
+    const server = setupServer(...getRegistrationInfoHandler(123455, getResponse));
+    before(() => server.listen());
+    afterEach(() => getResponse.reset());
+    afterEach(() => server.resetHandlers());
+    after(() => server.close());
+
+    it('returns data as requested', async () => {
+      getResponse.returns({
+        "code": 200,
+        "error": null,
+        "data": {
+          "id": 123455,
+          "accountId": 1311,
+          "registrationId": "593409403",
+          "transactionCount": 1
+        }
+      });
+      const result = await getRegistrationInfo(123455);
+
+      expect(result).to.include({ 'transactionCount': 1 });
+    });
+
+    it('handles a failure in the API', async () => {
+      getResponse.returns({
+        error: ['heky!'],
+      });
+      return getRegistrationInfo(123455).should.eventually.be.rejected;
```

Comment:
Something about super fluent APIs that makes me itchy over time. Reminds me of trying to write Cucumber syntax ages ago...

Reply:
I definitely don't like how these aren't methods. It feels weird to `.object.object.object` instead of `.method().method().method()`. But having said that, I /looooove/ fluent APIs! Chai comes with [Asserts](https://www.chaijs.com/api/assert/) and I won't stop you if you want to use them. But I'm using Chai's fluent expectations personally :D.